### PR TITLE
Feature/find lock by mac address

### PR DESCRIPTION
**Pull request by Marv05**
enabled finding lock by their mac_address by adding a function to lockrepo that tries to get the lock by its mac_address. you are still able to do this via lock_id

### CODE REVIEW

**Comment by Marv05** [comment]
and created unit tests!

File: app/Repositories/LockRepository.php [hunk 1]
```diff
@@ -61,10 +61,21 @@ public function getLock($lock_uuid): Lock
         try {
             return Lock::findOrFail($lock_uuid);
         } catch (Exception $e) {
+            return $this->tryToFindLockByMac($lock_uuid);
             throw new MyModelNotFoundException("Lock", $lock_uuid);
```
Comment (marfrede) [inline app/Repositories/LockRepository.php:65]: diese Zeile wird nie ausgeführt. Kann weg.

File: app/Repositories/LockRepository.php [hunk 2]
```diff
@@ -61,10 +61,21 @@ public function getLock($lock_uuid): Lock
...
         }
     }
 
+    private function tryToFindLockByMac($mac_address): Lock
```
Comment (marfrede) [inline app/Repositories/LockRepository.php:69]: better name: getLockByMac

**Review by marfrede — COMMENTED** [review]
1. Umbenennung getLockByMac
2. throw Exception in getLockById wird nicht ausgeführt. Führt zu schlechter Lesbarkeit. wenn get by id nicht klappt, dann catch und -> get by mac

File: app/Repositories/LockRepository.php (hunk 1 above)
Comment (Marv05) [reply app/Repositories/LockRepository.php:65]: upsa

**Pull request merged**